stringzilla/pybind11.cpp

```diff
@@ -521,12 +533,22 @@ PYBIND11_MODULE(stringzilla, m) {
     py_file.def("open", &py_file_t::open, py::arg("path"));
     py_file.def("open", &py_file_t::reopen);
     py_file.def("close", &py_file_t::close);
+    py_file.def("__getitem__", [](py_file_t &s, py::slice slice) {
+        py::ssize_t start, stop, step, slice_length;
+        slice.compute(s.size(), &start, &stop, &step, &slice_length);
+        return s.sub(start, stop);
+    });
 
     auto py_slices = py::class_<py_spans_t, std::shared_ptr<py_spans_t>>(m, "Slices");
     py_slices.def(py::init([]() { return std::make_shared<py_spans_t>(); }));
-    py_slices.def("sub", &py_spans_t::sub, py::arg("start") = 0, py::arg("end") = 0);
+    py_slices.def("sub", &py_spans_t::sub, py::arg("start") = 0, py::arg("end") = 0, py::arg("step") = 0);
     py_slices.def("__len__", &py_spans_t::size);
     py_slices.def("__getitem__", &py_spans_t::at, py::arg("index"));
+    py_slices.def("__getitem__", [](py_spans_t &s, py::slice slice) {
+        py::ssize_t start, stop, step, slice_length;
+        slice.compute(s.size(), &start, &stop, &step, &slice_length);
+        return s.sub(start, stop, step);
```

Comment:
This seems wrong. You are just dropping the `step`. Either report it or implement it.